lang/tags/comment_documentation.talon

```diff
@@ -1,3 +1,6 @@
 tag: user.code_comment_documentation
 -
 dock comment: user.code_comment_documentation()
+block dock comment: user.code_comment_documentation_block()
+inner dock comment: user.code_comment_documentation_inner()
+inner block dock comment: user.code_comment_documentation_block_inner()
```

Comment:
Are these kinds of commands universal enough to warrant putting them here? I guess block doc-comments are fairly common. I have no idea what an "inner dock comment" is.

Reply:
I'd probably say they're probably not really universal. Such a doc comment in Rust is most commonly used to add docs for a module within the module. (The syntax works in any block that can be documented, so `inner` would definitely be a correct name here.)

I'm not familiar with a similar construct in JavaScript, and docstrings in Python always go inside the block it documents, so :shrug:

Reply:
I suspect the concepts of a block comment is sufficiently generic, however I'm not too sure about inner document comments. In rust you can document and entity atop it with a doc comment, or you can document information about it on top of its relevant member with an inner doc comment. Is there a strong preference to have the inner cases removed here and integrated into the module inside the rust file? I thought it would be okay because I had seen functions inside tags only implemented by one language before, in the case of the R language.

Reply:
I think we'd prefer if these all were made Rust-specific, we can't find examples of other languages that have all four of these distinctions.